lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -1764,13 +1764,61 @@ bool TypeSystemSwiftTypeRef::IsFunctionPointerType(
   VALIDATE_AND_RETURN(impl, IsFunctionPointerType, type,
                       (ReconstructType(type)));
 }
+
 bool TypeSystemSwiftTypeRef::IsPossibleDynamicType(opaque_compiler_type_t type,
                                                    CompilerType *target_type,
                                                    bool check_cplusplus,
                                                    bool check_objc) {
-  return m_swift_ast_context->IsPossibleDynamicType(
-      ReconstructType(type), target_type, check_cplusplus, check_objc);
+  if (target_type)
+    target_type->Clear();
+
+  if (!type)
+    return false;
+
+  auto impl = [&]() {
+    using namespace swift::Demangle;
+    Demangler dem;
+    auto *node = DemangleCanonicalType(dem, type);
+    if (!node)
+      return false;
+
+    switch (node->getKind()) {
+    case Node::Kind::Class:
+    case Node::Kind::BoundGenericClass:
+    case Node::Kind::Protocol:
+    case Node::Kind::ProtocolList:
+    case Node::Kind::ProtocolListWithClass:
+    case Node::Kind::ProtocolListWithAnyObject:
+    case Node::Kind::DynamicSelf:
+      return true;
+    case Node::Kind::TypeAlias: {
+      if (node->getNumChildren() == 2) {
+        auto *module = node->getFirstChild();
+        auto *identifier = node->getLastChild();
+        return module->getKind() == Node::Kind::Module &&
+               identifier->getKind() == Node::Kind::Identifier &&
+               module->getText() == "Swift" &&
```

Comment:
nitpick: theres a swift::something_STDLIB_NAME constant in swift/strings.h that we should use here